modules/drivers/snowflake/src/metabase/driver/snowflake.clj

```diff
@@ -341,8 +341,9 @@
   accept either. Throw an Exception if neither key can be found."
   {:arglists '([database])}
   [{details :details}]
-  (or (:db details)
-      (:dbname details)
+  ;; ignore any blank keys
+  (or (m/find-first (every-pred string? (complement str/blank?))
+                    ((juxt :db :dbname) details))
```

Comment:
Not really needed for the fix, but while investigating the bug I realized that this wasn't ignoring blank strings like it should (just to be extra safe), so I fixed that here and added a test.